File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(-1, head)
        p = dummy
        q = dummy.next
        while q:
            q = q.next
            if not q:
                break
            q = q.next
            p = p.next
        p.next = p.next.next
        return dummy.next

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        ListNode* newHead = new ListNode(-1);
        newHead->next = head;
        head = newHead;
        ListNode* p = head;
        ListNode* q = head->next;
        while(q)
        {
            q = q->next;
            if(!q)
            {
                break;
            }
            q = q->next;
            p = p->next;
        }
        p->next = p->next->next;
        return head->next;
    }
};